src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,15 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        Optional<User> validUser = Optional.ofNullable(user);
+
+        if (validUser.isEmpty()) {
+            return false;
+        }
+
+        return validUser
+                .filter(inputUser -> inputUser.age != null && inputUser.age >= 30 && inputUser.age <= 45)
+                .isPresent();
```

Comment:
다음과 같이 체이닝 방식으로 작성해볼 수도 있을 것 같습니다.
```suggestion
        return Optional.ofNullable(user)
                .map(User::getAge)
                .filter(age -> age >= 30 && age <= 45)
                .isPresent();
```